src/parser.js

```diff
@@ -34,99 +46,171 @@ class Parser {
     }
 
-    if (isTextNode(node)) {
-      return `${indent}| ${node.value}`
-    } else if (isCommentNode(node)) {
-      return this.parseComment(node, indent)
-    } else if (isElementNode(node)) {
-      let line = `${indent}${this.setAttributes(node)}`
+    // Add CSS selectors to pug node and append any element attributes to it
+    for (const attr of attrs) {
+      const { name, value } = attr
 
-      if (this.isUniqueTextNode(node)) {
-        line += ` ${node.childNodes[0].value}`
+      // Remove div tag if a selector is present (shorthand)
+      // e.g. div#form() -> #form()
+      const hasSelector = name === 'id' || name === 'class'
+      if (tagName === DIV_NODE && hasSelector) {
+        pugNode = pugNode.replace(DIV_NODE, '')
       }
 
-      return line
-    } else {
-      return node
-    }
-  }
-
-  setAttributes (node) {
-    const { tagName, attrs: attributes } = node
-    const attributeList = []
-    let pugNode = tagName
-
-    attributes.forEach(({ name, value }) => {
-      let shorten = false
-
       switch (name) {
         case 'id':
-          shorten = true
           pugNode += `#${value}`
           break
         case 'class':
-          shorten = true
           pugNode += `.${value.split(' ').join('.')}`
           break
-        default:
-          attributeList.push(`${name}='${value}'`)
+        default: {
+          // Add escaped single quotes (\') to attribute values
+          const val = value.replace(/'/g, "\\'")
+          if (!val) {
+            attributes.push(name)
+          } else {
+            attributes.push(`${name}='${val}'`)
+          }
           break
+        }
       }
+    }
 
-      // Remove div tagName
-      if (tagName === 'div' && shorten) {
-        pugNode = pugNode.replace('div', '')
-      }
-    })
-
-    if (attributeList.length) {
-      pugNode += `(${attributeList.join(', ')})`
+    if (attributes.length) {
+      pugNode += `(${attributes.join(', ')})`
     }
 
     return pugNode
   }
 
-  isUniqueTextNode (node) {
-    return node.childNodes.length === 1 && isTextNode(node.childNodes[0])
+  /**
+   * formatPugNode applies the correct indent for the current line,
+   * and formats the value as either as a single or multiline string.
+   *
+   * @param {String} node - The pug node (e.g. header(class='foo'))
+   * @param {String} value - The node's value
+   * @param {Number} level - Current tree level to generate indent
+   * @param {String} blockChar - The character used to denote a multiline value
+   */
+  formatPugNode(node, value = '', level, blockChar = '.') {
+    const indent = this.getIndent(level)
+    const result = `${indent}${node}`
+
+    const lines = value.split('\n')
+
+    if (lines.length > 1) {
+      const indentChild = this.getIndent(level + 1)
+      const multiline = lines.map(line => `${indentChild}${line}`).join('\n')
+      return `${result}${blockChar}\n${multiline}`
+    }
+
+    return value.length ? `${result} ${value}` : result
```

Comment:
you could return much earlier in case value.lengh === 0

Reply:
and not have ternary

Reply:
You're right, you can return early here by inverting the `if` statement. However, you still need the ternary to avoid trailing whitespace in case of an empty value.